samples/extauthz/src/main.go

```diff
@@ -71,7 +73,7 @@ func (s *extAuthzServerV2) Check(ctx context.Context, request *authv2.CheckReque
 		request.GetAttributes().GetRequest().GetHttp().GetHost(),
 		request.GetAttributes().GetRequest().GetHttp().GetPath(),
 		request.GetAttributes())
-	if allowedValue == request.GetAttributes().GetRequest().GetHttp().GetHeaders()[checkHeader] {
+	if allowedValue == request.GetAttributes().GetRequest().GetHttp().GetHeaders()[checkHeader] || strings.HasSuffix(request.GetAttributes().Source.Principal, *serviceaccount) {
```

Comment:
you should parse the principal and only check the last segment after "/" and you also need to change line 121 for gRPC v3.

Reply:
what about HasSuffix("/"+serviceAccount) ? Do you think this is a feasible way?